py/desitarget/mock/mockmaker.py

```diff
@@ -368,30 +405,113 @@ def imaging_depth(self, data):
             data['PSFDEPTH_{}'.format(band)] = np.repeat(psfdepth_ivar[ii], nobj)
             data['GALDEPTH_{}'.format(band)] = np.repeat(galdepth_ivar[ii], nobj)
 
-        # compute the WISE depth, which is largely a function of ecliptic latitude 
-        coord = SkyCoord(data['RA']*u.deg, data['DEC']*u.deg)
-        ecoord = coord.transform_to('barycentrictrueecliptic')
-        beta = ecoord.lat.value
-        if np.count_nonzero(beta > 89) > 0: # don't explode at the pole!
-            beta[beta > 89] = 89
-        if np.count_nonzero(beta < -89) > 0: # don't explode at the pole!
-            beta[beta < -89] = -89
-        beta = np.radians(ecoord.lat.value) # [radians]
+        self.set_wise_depth(data)
 
-        sig_syst = [0.5, 2.0]                   # systematic uncertainty due to low-level 
-                                                # background structure e.g. striping
-        neff = [15.7832, 18.5233]               # effective number of pixels in PSF
-        vega2ab = [2.699, 3.339]
-        sig_stat_beta0 = [3.5127802, 9.1581879] # random uncertainty [AB nanomaggies]
+    def imaging_depth(self, data, legacy_dir=None, aprad=0.0):
+        import  desitarget.randoms as     randoms
 
-        for ii, band in enumerate(('W1', 'W2')):
-            sig_stat = sig_stat_beta0[ii] / np.sqrt( 1.0 / np.cos(beta) )
-            sig = np.sqrt( sig_stat**2 + sig_syst[ii]**2 )
+        from    desitarget.targets import resolve
+        from    astropy.table      import Table
+        from    desitarget.io      import basetsdatamodel
 
-            wisedepth_mag = 22.5 - 2.5 * np.log10( sig * np.sqrt(neff[ii]) ) + vega2ab[ii] # 1-sigma, AB mag
-            wisedepth_ivar = 1 / (5 * 10**(-0.4 * (wisedepth_mag - 22.5)))**2 # 5-sigma, 1/nanomaggies**2
-            data['PSFDEPTH_{}'.format(band)] = wisedepth_ivar
+        if legacy_dir is not None:
+            log.info('Populating image depths & maskbits.')
+            
+            bricks      = self.Bricks
 
+            # Return brick name of brick covering (ra, dec).
+            bricknames  = bricks.brickname(data['RA'], data['DEC'])
+
+            ubricknames = np.unique(bricknames)
+
+            # All bricks potentially available in this release.  If not present,
+            # code has exited before targets_truth call.
+            standard    = fitsio.read(legacy_dir + '/survey-bricks.fits.gz')
+            standard    = np.char.decode(standard['BRICKNAME'], 'UTF-8')
+            
+            #
+            keep        = ['MASKBITS', 'PHOTSYS']  
+            
+            bands       = ['G', 'R', 'Z']
+            bandkeep    = ['NOBS', 'PSFDEPTH', 'GALDEPTH']
+        
+            _           = empty_targets_table(nobj=1)
+            dtypes      = dict(zip(_.columns, [_[x].dtype for x in _.columns]))
```

Comment:
I hate to be nitpicky but the offset/column-aligned equals signs (among other changes) is a significant change in style relative to the surrounding code. A good rule is to use the style of the existing code, while new modules can use your own personal style (although DESI code *generally* tries to conform to the PEP8 guidelines).

Reply:
It's clear there's going to be significant changes to this to get it into desihub and fair enough, this doesn't need to be one of them.  This was my version and I find it helps me parse dense parts of the code.